### PR TITLE
Typo at perlfaq8

### PR DESCRIPTION
**Pull request by joaquinferrero**

### CODE REVIEW

**Comment by karenetheridge** [comment]
Please could you rebase that to remove the merge commits, and collapse like changes together?

**Comment by joaquinferrero** [comment]
Sorry, Karen, but... I don't know how I can to rebase this mess. Any tip?

**Comment by karenetheridge** [comment]
Make sure you've got the updated master pulled down, then check out your branch and `git rebase master`. Then force-push back up to github and the pull request will automatically update.  If you're not used to rebasing, I suggest checking out some tutorials - interactive rebasing in particular is really awesome, and I rely on it heavily when developing in multiple branches at once.

File: lib/perlfaq6.pod [hunk 1]
```diff
@@ -114,7 +114,7 @@ be mangled by many mailers):
     $/ = '';          # read in whole paragraph, not just one line
     while ( <> ) {
         while ( /^From /gm ) { # /m makes ^ match next to \n
-        print "leading From in paragraph $.\n";
+        print "leading from in paragraph $.\n";
```
Comment (karenetheridge) [inline lib/perlfaq6.pod]: The text being matched is "From", so it was correct before.

File: lib/perlfaq5.pod [hunk 2]
```diff
@@ -1539,7 +1539,7 @@ a similar interface, but does the traversal for you too:
 (contributed by brian d foy)
 
 If you have an empty directory, you can use Perl's built-in C<rmdir>.
-If the directory is not empty (so, with files or subdirectories), you
+If the directory is not empty (so, no files or subdirectories), you
```
Comment (karenetheridge) [inline lib/perlfaq5.pod]: This was just corrected (by you) in commit 9f411960. Why is it being reverted again?

File: lib/perlfaq6.pod (hunk 1 above)
Comment (joaquinferrero) [reply lib/perlfaq6.pod]: Ok

File: lib/perlfaq5.pod (hunk 2 above)
Comment (joaquinferrero) [reply lib/perlfaq5.pod]: Sorry, Karen. I mess all this pull.. Sorry. I will try to repair it.

Comment (joaquinferrero) [reply lib/perlfaq5.pod]: Karen, Can you remove all this commits/branch? Next, I can to resend all the changes into a new pull request.

Comment (karenetheridge) [reply lib/perlfaq5.pod]: No, you can fix it all yourself in this branch.  https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History  This is a very important skill to learn.

**Comment by karenetheridge** [comment]
To rebase, this is what I would do:

- start in the master branch. `git checkout master`
- you made your changes right in master (not a good idea); let's move them to their own branch. `git checkout -b jf/typos`
- make sure that the latest changes from upstream are pulled down (you should have this remote already, but probably named something else.) `git checkout master; git remote add upstream git@github.com:perl-doc-cats/perlfaq.git; git reset --hard upstream/master`
- go into your pull request branch and bring the commits from master into it, and clean them up. `git checkout jf/typos; git rebase -i master`
- this is where the tutorial comes in. you are now in an editor and you can move around your commits, squash them together, etc. I suspect that they should all be collapsed together into one commit, but I haven't looked closely. To do that, change all lines but the first from `pick` to `squash`, and save/exit the editor.  Now you're in the editor for the commit message - edit all the text down to something short and descriptive, like 'fix typos'. Save/quit.
- use `git log --oneline --graph --decorate` and `git log -p` to confirm that you are up-to-date with master, there are no merge commits remaining, and your desired changes are still there.
- when ready, `git push -u origin jf/typos`, and come to this pull request and edit the branch that the pull request comes from. Never use master for pull requests - it makes it harder to manage multiple changes at once!

File: lib/perlfaq5.pod (hunk 2 above)
Comment (joaquinferrero) [reply lib/perlfaq5.pod]: Maybe the problem that I edited & pull the changes with github only. I don't have a cloned git repository of perlfaq at my computer. I use the option of github of "edit this file" (in other words, fork the perlfaq repository, edit file(s), create pull request, all within github web pages).

**Comment by karenetheridge** [comment]
Ok. I rebased it locally and then merged it. thanks!

**Pull request closed without merging**